operators/pkg/controller/elasticsearch/dynamic_watches.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
I would move that into `elasticsearch/certificates/http/watch.go`. It feels a bit weird to have it at the top-level. For instance we also have dynamic watches for secure settings in `elasticsearch/settings`.